Zend/zend_fibers.c

```diff
@@ -237,50 +253,30 @@ ZEND_API void zend_fiber_switch_context(zend_fiber_context *to)
 		to->stack.prior_size);
 #endif
 
-	EG(current_fiber) = to;
-
 	transfer_t transfer = jump_fcontext(to->handle, from);
-	((zend_fiber_context *) transfer.data)->handle = transfer.context;
-
-	EG(current_fiber) = from;
 
 #ifdef __SANITIZE_ADDRESS__
 	__sanitizer_finish_switch_fiber(fake_stack, &to->stack.prior_pointer, &to->stack.prior_size);
 #endif
-}
 
-static void zend_fiber_suspend_from(zend_fiber *fiber)
-{
-	zend_fiber_vm_state state;
-
-	ZEND_ASSERT(fiber->caller && "Fiber has no caller");
-
-	zend_fiber_capture_vm_state(&state);
-	zend_fiber_switch_context(zend_fiber_get_context(fiber->caller));
-	zend_fiber_restore_vm_state(&state);
-}
-
-static void zend_fiber_switch_to(zend_fiber *fiber)
-{
-	zend_fiber_context *context = zend_fiber_get_context(fiber);
-	zend_fiber_vm_state state;
+	((zend_fiber_context *) transfer.data)->handle = transfer.context;
 
-	zend_observer_fiber_switch_notify(EG(current_fiber), context);
+	EG(current_fiber) = from;
 
-	fiber->caller = zend_fiber_from_context(EG(current_fiber));
+	if (to->caller == from && to->status == ZEND_FIBER_STATUS_RUNNING) {
```

Comment:
Should we consider any fiber that isn't the current fiber as suspended? Otherwise this code might be fragile if an extension didn't treat fibers like a stack.